### PR TITLE
New versions of Assign/ReadValue ops

### PR DESCRIPTION
**Pull request by itikhono**
- new versions Assign/ReadValue operations in opset6 (ticket 32173)
- implemented a base class for Assign/ReadValue ops
- updated LowLatency transformation for Loop op (ticket 47796)
- updated tests

**note:** unit tests for opset6 versions of Assign/ReadValue ops will be added in the next PR to simplify the review process
MO e2e tests cover necessary functionality in this PR

### CODE REVIEW

File: ngraph/core/include/ngraph/op/assign.hpp [hunk 1]
```diff
@@ -23,10 +23,33 @@ namespace ngraph
...
+        {
+        public:
+            NGRAPH_RTTI_DECLARATION;
+            AssignBase() = default;
+            /// \brief Constructs an AssignBase operation.
+            explicit AssignBase(const OutputVector& arguments)
+                : Sink(arguments)
+            {
+            }
+            virtual std::shared_ptr<ngraph::Variable> get_variable() const { return m_variable; }
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/assign.hpp:36]: Please add doxygen comment to all public methods.

File: ngraph/core/src/op/assign.cpp [hunk 2]
```diff
@@ -91,3 +94,32 @@ bool op::v3::Assign::visit_attributes(AttributeVisitor& visitor)
...
+op::v6::Assign::Assign(const Output<Node>& new_value, const std::shared_ptr<Variable>& variable)
+    : AssignBase({new_value})
+{
+    m_variable = variable;
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::Assign::validate_and_infer_types()
+{
+    m_variable->update({get_input_partial_shape(0),
```
Comment (ilyachur) [inline ngraph/core/src/op/assign.cpp:107]: Please add CC macro

File: ngraph/core/src/op/read_value.cpp [hunk 3]
```diff
@@ -56,3 +58,43 @@ bool op::v3::ReadValue::visit_attributes(AttributeVisitor& visitor)
...
+    NGRAPH_OP_SCOPE(v6_ReadValue_validate_and_infer_types);
+    const auto arg_t = get_input_element_type(0);
+    auto output_shape = get_input_partial_shape(0);
+    NGRAPH_CHECK(m_variable, "Variable is not initialized.");
+    VariableInfo var_info = {output_shape, element::dynamic, m_variable->get_info().variable_id};
+    NODE_VALIDATION_CHECK(
+        this,
+        element::Type::merge(var_info.data_type, m_variable->get_info().data_type, arg_t),
+        "Variables types are inconsistent.");
+    /*    NODE_VALIDATION_CHECK(
```
Comment (ilyachur) [inline ngraph/core/src/op/read_value.cpp:80]: Why do we have a commented code?

Comment (itikhono) [reply ngraph/core/src/op/read_value.cpp]: fixed

File: ngraph/core/src/op/assign.cpp (hunk 2 above)
Comment (itikhono) [reply ngraph/core/src/op/assign.cpp:108]: fixed

File: ngraph/core/include/ngraph/op/assign.hpp (hunk 1 above)
Comment (itikhono) [reply ngraph/core/include/ngraph/op/assign.hpp:38]: fixed

**Review by sadolini — APPROVED** [review]
LGTM

**Comment by GlebKazantaev** [comment]
@itikhono @sadolini do you have in plans to update docs with Assign/Read/Variable explanations?

**Comment by itikhono** [comment]
> @itikhono @sadolini do you have in plans to update docs with Assign/Read/Variable explanations?

@GlebKazantaev  We have the page in docs describing the general mechanism of Memory in OpenVINO https://docs.openvinotoolkit.org/latest/openvino_docs_IE_DG_network_state_intro.html 
I will update it according to these changes.

**Review by GlebKazantaev — APPROVED** [review]
In general LGTM.
Can we add IR Serialization tests for Read/Assign ops? inference-engine/tests/functional/inference_engine/ir_serialization
Just to double check that the whole pipeline works (can be done in separatePR).

**Review by ilyachur — APPROVED** [review]

**Comment by itikhono** [comment]
> In general LGTM.
> Can we add IR Serialization tests for Read/Assign ops? inference-engine/tests/functional/inference_engine/ir_serialization
> Just to double check that the whole pipeline works (can be done in separatePR).

Sure, I will add these tests!

**Pull request merged**